auth/src/main/java/org/aerogear/mobile/auth/AuthService.java

```diff
@@ -204,6 +204,10 @@ public void init(final Context context,
                             "configure method must be called before the init method");
         }
 
+        CertificatePinningCheck pinningCheck =
+                        new CertificatePinningCheck(this.mobileCore.getHttpLayer());
+        pinningCheck.check(this.serviceConfiguration.getUrl());
```

Comment:
Can this check be done in an earlier step of the build configuration process?  (For example when the mobile-services.json file is parsed).

Also what does cert pinning check do and how is it supposed to be used?

Reply:
I don't think the check should happen at this point. Also if there are some other errors (e.g. the device is offline at the time when the check is happening), the login will not work and there is no way to reset it unless the app is restarted. I don't think that is the expected behaviour. If I remember correctly, this check is only required for the login request, right before the browser is being opened. So I think this should be done right before the tab is opened.